### PR TITLE
influxdb: Add documentation for the 2.x line.

### PR DESCRIPTION
**Pull request by danxmoran**
See docker-library/official-images#9594

### CODE REVIEW

File: influxdb/content.md [hunk 1]
```diff
@@ -6,32 +6,395 @@ InfluxDB is a time series database built from the ground up to handle high write
...
+      %%IMAGE%%
+```
+
+This will make the generated CLI configs available to the host.
+
+#### Automated Setup
+
+The InfluxDB image contains some extra functionality to automatically bootstrap the system. This functionality is enabled by setting the `INFLUXD_INIT_MODE` environment variable to the value `setup` when running the container. Additional environment variables are used to configure the setup logic:
+
+-	`INFLUXD_INIT_USERNAME`: The username to set for the system's initial super-user (**Required**).
```
Comment (danxmoran) [inline influxdb/content.md:144]: Caught by an early adopter: all of these env variables need corrected, the prefix is `INFLUXDB_`. Will fix

File: influxdb/content.md [hunk 2]
```diff
@@ -6,32 +6,439 @@ InfluxDB is a time series database built from the ground up to handle high write
 
 %%LOGO%%
 
-## Updating latest to 2.0
+## `latest` updated to 2.0
 
-In the future, the latest tag for this image will point to the latest released implementation of influxdb 2.0. This will happen when the first general available release for 2.0 exists. If you are using the `latest` tag for any production or development purposes, please update your development environment to reference the `1.8` tag.
+The `latest` tag for this image now points to the latest released implementation of InfluxDB 2.0. If you are using the `latest` tag and would like to stay on the 1.x line, please update your environment to reference the `1.8` tag.
```
Comment (timhallinflux) [inline influxdb/content.md:11]: 2.x will be more durable.....

File: influxdb/content.md [hunk 3]
```diff
@@ -6,32 +6,439 @@ InfluxDB is a time series database built from the ground up to handle high write
...
+      %%IMAGE%% --storage-wal-fsync-delay=15m
+```
+
+CLI options take precedence over environment variables.
+
+Find more about configuring InfluxDB [here](https://docs.influxdata.com/influxdb/v2.0/reference/config-options/).
+
+### Database Setup
+
+InfluxDB 2.0 requires authentication. A special API exists to bootstrap the first super-user in the database, along with an initial organization and bucket. It's possible to access this API manually, or to run it automatically via environment variables.
```
Comment (timhallinflux) [inline influxdb/content.md:133]: again ... 2.x here.

File: influxdb/content.md [hunk 4]
```diff
@@ -6,32 +6,439 @@ InfluxDB is a time series database built from the ground up to handle high write
...
+      -e INFLUXDB_INIT_ORG=my-org \
+      -e INFLUXDB_INIT_BUCKET=my-bucket \
+      -e INFLUXDB_INIT_RETENTION=1w \
+      -e INFLUXDB_INIT_ADMIN_TOKEN=my-super-secret-auth-token \
+      %%IMAGE%%
+```
+
+**NOTE:** Automated setup will not run if an existing boltdb file is found at the configured path. This behavior allows for the InfluxDB container to reboot post-setup without encountering "DB is already set up" errors.
+
+### Interacting with the DB
```
Comment (timhallinflux) [inline influxdb/content.md:229]: Interacting with InfluxDB

File: influxdb/content.md [hunk 5]
```diff
@@ -6,32 +6,439 @@ InfluxDB is a time series database built from the ground up to handle high write
...
+
+Once an InfluxDB instance has completed initial setup, its APIs will unlock. See the main documentation site for reference information and examples on:
+
+-	[Writing data](https://docs.influxdata.com/influxdb/v2.0/write-data/)
+-	[Reading data](https://docs.influxdata.com/influxdb/v2.0/query-data/)
+-	[Configuring security](https://docs.influxdata.com/influxdb/v2.0/security/)
+-	[And more!](https://docs.influxdata.com/influxdb/v2.0/)
+
+### Upgrading from 1.x
```
Comment (timhallinflux) [inline influxdb/content.md:238]: InfluxDB 1.x

File: influxdb/content.md [hunk 6]
```diff
@@ -6,32 +6,439 @@ InfluxDB is a time series database built from the ground up to handle high write
...
+
+InfluxDB 2.0 provides a 1.x-compatible API, but expects a different storage layout on disk. To bridge this mismatch, the InfluxDB image contains extra functionality to migrate 1.x data and config into 2.x layouts automatically before booting the `influxd` server.
```
Comment (timhallinflux) [inline influxdb/content.md:240]: 2.x

File: influxdb/content.md [hunk 7]
```diff
@@ -6,32 +6,439 @@ InfluxDB is a time series database built from the ground up to handle high write
...
+
+Automated upgrade will generate both data and config files, by default under `/var/lib/influxdb2` and `/etc/influxdb2`. It's recommended to mount volumes at both paths to avoid losing data.
+
+**NOTE:** Automated upgrade will not run if an existing boltdb file is found at the configured path. This behavior allows for the InfluxDB container to reboot post-upgrade without overwriting migrated data.
+
+Find more about the InfluxDB upgrade process [here](https://docs.influxdata.com/influxdb/v2.0/upgrade/v1-to-v2/). See below for examples of common upgrade scenarios.
+
+#### Upgrade Example - Minimal
+
+Assume you've been running a minimal 1.x deployment:
```
Comment (timhallinflux) [inline influxdb/content.md:273]: InfluxDB 1.x

File: influxdb/content.md [hunk 8]
```diff
@@ -6,32 +6,439 @@ InfluxDB is a time series database built from the ground up to handle high write
...
+
+```console
+$ docker run -p 8086:8086 \
+      -v influxdb:/var/lib/influxdb \
+      %%IMAGE%%:1.8
+```
+
+To upgrade this deployment to 2.x, stop the running 1.x container, then run:
```
Comment (timhallinflux) [inline influxdb/content.md:281]: InfluxDB 2.x and InfluxDB 1.x

File: influxdb/content.md [hunk 9]
```diff
@@ -6,32 +6,439 @@ InfluxDB is a time series database built from the ground up to handle high write
...
+      -v influxdb2:/var/lib/influxdb2 \
+      -e INFLUXDB_INIT_MODE=upgrade \
+      -e INFLUXDB_INIT_USERNAME=my-user \
+      -e INFLUXDB_INIT_PASSWORD=my-password \
+      -e INFLUXDB_INIT_ORG=my-org \
+      -e INFLUXDB_INIT_BUCKET=my-bucket \
+      %%IMAGE%%
+```
+
+#### Upgrade Example - Custom 1.x Config
```
Comment (timhallinflux) [inline influxdb/content.md:295]: Custom InfluxDB 1.x Config

File: influxdb/content.md [hunk 10]
```diff
@@ -6,32 +6,439 @@ InfluxDB is a time series database built from the ground up to handle high write
...
+
+Assume you've been running a 1.x deployment with customized config:
```
Comment (timhallinflux) [inline influxdb/content.md:297]: InfluxDB 1.x

File: influxdb/content.md [hunk 11]
```diff
@@ -6,32 +6,439 @@ InfluxDB is a time series database built from the ground up to handle high write
...
+
+```console
+$ docker run -p 8086:8086 \
+      -v influxdb:/var/lib/influxdb \
+      -v $PWD/influxdb.conf:/etc/influxdb/influxdb.conf:ro \
+      %%IMAGE%%:1.8
+```
+
+To upgrade this deployment to 2.x, stop the running container, then run:
```
Comment (timhallinflux) [inline influxdb/content.md:306]: InfluxDB 2.x

File: influxdb/content.md [hunk 12]
```diff
@@ -6,32 +6,439 @@ InfluxDB is a time series database built from the ground up to handle high write
...
+      -e INFLUXDB_INIT_USERNAME=my-user \
+      -e INFLUXDB_INIT_PASSWORD=my-password \
+      -e INFLUXDB_INIT_ORG=my-org \
+      -e INFLUXDB_INIT_BUCKET=my-bucket \
       %%IMAGE%%
 ```
 
+#### Upgrade Example - Custom Paths
+
+Assume you've been running a 1.x deployment with data and config mounted at custom paths:
```
Comment (timhallinflux) [inline influxdb/content.md:323]: InfluxDB 1.x

File: influxdb/content.md [hunk 13]
```diff
@@ -6,32 +6,439 @@ InfluxDB is a time series database built from the ground up to handle high write
...
+
+```console
+$ docker run -p 8086:8086 \
+      -v influxdb:/root/influxdb/data \
+      -v $PWD/influxdb.conf:/root/influxdb/influxdb.conf:ro \
+      %%IMAGE%%:1.8 -config /root/influxdb/influxdb.conf
+```
+
+To upgrade this deployment to 2.x, first decide if you'd like to keep using custom paths, or use the 2.x defaults. If you decide to use the defaults, you'd stop the running 1.x container, then run:
```
Comment (timhallinflux) [inline influxdb/content.md:332]: add InfluxDB in front of all 1.x and 2.x versions

File: influxdb/content.md [hunk 14]
```diff
@@ -6,32 +6,439 @@ InfluxDB is a time series database built from the ground up to handle high write
...
+
+As a convenience for script-writers, the image will export a number of variables into the environment before executing any scripts:
+
+-	`INFLUX_CONFIGS_PATH`: Path to the CLI configs file written by `setup`/`upgrade`
+-	`INFLUX_HOST`: URL to the `influxd` instance running setup logic
+-	`INFLUXDB_INIT_USER_ID`: ID of the initial admin user created by `setup`/`upgrade`
+-	`INFLUXDB_INIT_ORG_ID`: ID of the initial organization created by `setup`/`upgrade`
+-	`INFLUXDB_INIT_BUCKET_ID`: ID of the initial bucket created by `setup`/`upgrade`
+
+For example, if you wanted to grant write-access to a 1.x client on your initial bucket, you'd first create the file `$PWD/scripts/setup-v1.sh` with contents:
```
Comment (timhallinflux) [inline influxdb/content.md:379]: InfluxDB 1.x client

File: influxdb/content.md [hunk 15]
```diff
@@ -6,32 +6,439 @@ InfluxDB is a time series database built from the ground up to handle high write
...
+      -e V1_DB_NAME=v1-db \
+      -e V1_RP_NAME=v1-rp \
+      -e V1_AUTH_USERNAME=v1-user \
+      -e V1_AUTH_PASSWORD=v1-password \
+      %%IMAGE%%
+```
+
+**NOTE:** Custom scripts will not run if an existing boltdb file is found at the configured path (causing `setup` or `upgrade` to be skipped). This behavior allows for the InfluxDB container to reboot post-initialization without encountering errors from non-idempotent script commands.
+
+## Using this Image - 1.x
```
Comment (timhallinflux) [inline influxdb/content.md:420]: InfluxDB 1.x

File: influxdb/variant-data.md [hunk 16]
```diff
@@ -2,6 +2,6 @@
 
 *This image requires a valid license key from InfluxData.* Please visit our [products page](https://www.influxdata.com/products/) to learn more.
 
-This image contains the enterprise data node package for clustering. It supports all of the same options as the OSS image, but it needs port 8088 to be exposed to the meta nodes.
+This image contains the enterprise data node package for clustering. It supports all of the same options as the 1.x OSS image, but it needs port 8088 to be exposed to the meta nodes.
```
Comment (timhallinflux) [inline influxdb/variant-data.md:5]: InfluxDB 1.x OSS image

File: influxdb/content.md [hunk 17]
```diff
@@ -6,32 +6,439 @@ InfluxDB is a time series database built from the ground up to handle high write
...
+docker run -p 8086:8086 \
+      -e INFLUXD_BOLT_PATH=/root/.influxdbv2/influxd.bolt \
+      -e INFLUXD_ENGINE_PATH=/root/.influxdbv2/engine \
+      -v $PWD:/root/.influxdbv2 \
+      %%IMAGE%%
+```
+
+See the section about configuration below for more ways to override the data paths.
+
+## Using this Image - 2.x
```
Comment (timhallinflux) [inline influxdb/content.md:57]: InfluxDB 2.x

File: influxdb/content.md [hunk 18]
```diff
@@ -6,32 +6,439 @@ InfluxDB is a time series database built from the ground up to handle high write
...
 %%LOGO%%
 
-## Updating latest to 2.0
+## `latest` updated to 2.0
 
-In the future, the latest tag for this image will point to the latest released implementation of influxdb 2.0. This will happen when the first general available release for 2.0 exists. If you are using the `latest` tag for any production or development purposes, please update your development environment to reference the `1.8` tag.
+The `latest` tag for this image now points to the latest released implementation of InfluxDB 2.0. If you are using the `latest` tag and would like to stay on the 1.x line, please update your environment to reference the `1.8` tag.
 
-## Using this Image
+## Upgrading from quay.io-hosted 2.x
```
Comment (timhallinflux) [inline influxdb/content.md:13]: InfluxDB 2.x image

**Review by timhallinflux — APPROVED** [review]

**Review by tianon — APPROVED** [review]

**Pull request merged**